state/protocol/badger/snapshot.go

```diff
@@ -419,12 +419,12 @@ func (q *EpochQuery) Next() protocol.Epoch {
 	// if we are in setup phase, return a SetupEpoch
 	nextSetup := entry.NextEpochSetup
 	if phase == flow.EpochPhaseSetup {
-		return inmem.NewSetupEpoch(nextSetup)
+		return inmem.NewSetupEpoch(nextSetup, entry.NextEpoch.EpochExtensions)
```

Comment:
would suggest adding TODO to indicate that epochs will only be exposed once they are committed.